cou/utils/juju_utils.py

```diff
@@ -509,3 +520,304 @@ def _check_time(self) -> None:
             raise TimeoutException(
                 f"Model {self.model_name} has not stabilized after {self.timeout} seconds."
             )
+
+
+def retry(
+    function: Optional[Callable] = None,
```

Comment:
I was wondering why do we need to accept `None` as a possible value for the `function` parameter?

Reply:
Because if you want to use decorator without any configuration or with some additional configuration. 
example:
```python
>>> def decorator(func, debug=False):
...  def wrapper(*args, **kwargs):
...   print("decorated function", func.__name__, debug)
...   return func(*args, **kwargs)
...  return wrapper
... 
>>> @decorator
... def test1():
...  print("test1")
... 
>>> @decorator(debug=True)
... def test2():
...  print("test2")
... 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: decorator() missing 1 required positional argument: 'func'
>>> test1()
decorated function test1 False
test1
```

Reply:
I see, thanks for the explanation.